src/distance/JointAvoidanceConfigurationRanker.cpp

```diff
@@ -0,0 +1,58 @@
+#include "aikido/distance/JointAvoidanceConfigurationRanker.hpp"
+
+namespace aikido {
+namespace distance {
+
+using statespace::dart::ConstMetaSkeletonStateSpacePtr;
+using ::dart::dynamics::ConstMetaSkeletonPtr;
+
+//==============================================================================
+JointAvoidanceConfigurationRanker::JointAvoidanceConfigurationRanker(
+    ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+    ConstMetaSkeletonPtr metaSkeleton)
+  : ConfigurationRanker(
+        std::move(metaSkeletonStateSpace), std::move(metaSkeleton))
+  , mLowerLimitsState(mMetaSkeletonStateSpace->createState())
+  , mUpperLimitsState(mMetaSkeletonStateSpace->createState())
+{
+  auto lowerLimits = mMetaSkeleton->getPositionLowerLimits();
+  auto upperLimits = mMetaSkeleton->getPositionUpperLimits();
+
+  for (std::size_t i = 0; i < mMetaSkeletonStateSpace->getDimension(); ++i)
```

Comment:
Quick question: is there ever a case where there are lower limits but not upper limits (or vice versa)? If not, then rather than checking against `inf`s, maybe we can use `DegreeOfFreedom::hasPositionLimit`. I think it'd look something like this:
```cpp
auto properties = mMetaSkeletonStateSpace->getProperties();
for (std::size_t i = 0; i < properties.getNumDofs(); ++i)
{
  if (mMetaSkeleton->getDof(i)->hasPositionLimit())
    mUnboundedIndices.emplace_back(i);
}
```